### DISCUSSION TITLE
Way to view forks of sketches?

### DISCUSSION THREAD

**Discussion by jywarren** [processing/p5.js-web-editor#2659] [General]
#### Nature of issue?

- New feature request

#### New feature details:

Not sure if this exists or is under discussion but I'm happy to join a thread if so; I was curious about the "lineage" of sketches I "fork" from others' sketches -- it auto-copies the sketch when i start editing and save someone else's sketch, which is great... is there a way to either:

1. see what other sketch a given sketch was copied from originally?
2. see what sketches were copied from a given sketch?

Thanks!

**Comment by catarak** [comment]
thanks for reporting! i feel like this discussion has come up before, though i can't find it in any issues, so maybe it was in person or over email. 

in theory, this isn't too difficult to add. every sketch would need an array of sketches of duplicates, and a sketch could have a parent sketch, and then, a UI design/implementation.

when do you find yourself looking for the parent sketch/child sketches? i'm curious to find a larger "user story" that this fits into, or whether this would be more useful as something that would be returned by a future API endpoint.

**Comment by jywarren** [comment]
Hi, thank you for your swift response! Yes, the use-case is a workshop
where I'd like people to clone a sketch (a version of what I'd shown you a
couple years ago, browser-native arduino sensor connection:
https://editor.p5js.org/jywarren/sketches/rkztwSt8M) and build on it as an
example, but a) to help them be able to see others' forks to see the source
code for fellow attendees' work, and b) to have a "gallery" of sketches
from the workshop without having to manually gather URLs or usernames.

Kind of like how Scratch shows "remixes" upstream and downstream!

Thank you!

**Comment by catarak** [comment]
got it! a feature that will be added soon is collections of sketches (which could be from one account or multiple accounts, though still figuring out how to make the latter work), so that could also be helpful.

**Comment by jywarren** [comment]
oh awesome. That'd be great. And a collections system could in theory, in
the future, be a place to aggregate something in a more automated way... ok
cool! Thanks!
